dask/array/routines.py

```diff
@@ -524,32 +525,41 @@ def gradient(f, *varargs, **kwargs):
 
 
 @wraps(np.bincount)
-def bincount(x, weights=None, minlength=None):
-    if minlength is None:
-        raise TypeError("Must specify minlength argument in da.bincount")
+def bincount(x, weights=None, minlength=0):
     assert x.ndim == 1
     if weights is not None:
         assert weights.chunks == x.chunks
 
-    # Call np.bincount on each block, possibly with weights
     token = tokenize(x, weights, minlength)
-    name = 'bincount-' + token
-    if weights is not None:
-        dsk = {(name, i): (np.bincount, (x.name, i), (weights.name, i), minlength)
-               for i, _ in enumerate(x.__dask_keys__())}
-        dtype = np.bincount([1], weights=[1]).dtype
+    if minlength != 0:
+        dsk = {('minlength-' + token, 0): minlength}
```

Comment:
Keys can also be just strings.  It might be natural to convert `('minlength-123', 0)` to `'minlength-123'`

Reply:
I couldn't work out how to mix the two types (tuple keys, and string keys) in the one dask computation dictionary! I'm still not sure I really understand the higher level graph objects.